includes/collection/class-actors.php

```diff
@@ -389,4 +389,247 @@ public static function get_type_by_id( $user_id ) {
 
 		return 'user';
 	}
+
+	/**
+	 * Create or update a remote Actor (e.g., a follower) as a custom post type.
+	 *
+	 * @param array|Actor $actor_data The ActivityPub actor object as associative array (must include 'id').
+	 *
+	 * @return int|\WP_Error The post ID or WP_Error.
+	 */
+	public static function add( $actor_data ) {
+		if ( \is_array( $actor_data ) ) {
+			$actor_data = Actor::init_from_array( $actor_data );
+		}
+
+		if ( ! $actor_data instanceof Actor ) {
+			return new \WP_Error(
+				'activitypub_invalid_actor_data',
+				\__( 'Invalid actor data', 'activitypub' ),
+				array( 'status' => 400 )
+			);
+		}
+
+		$inbox = null;
+		if ( ! empty( $actor_data->get_endpoints()['sharedInbox'] ) ) {
+			$inbox = $actor_data->get_endpoints()['sharedInbox'];
+		} elseif ( ! empty( $actor_data->get_inbox() ) ) {
+			$inbox = $actor_data->get_inbox();
+		}
```

Comment:
Is there a chance `$inbox ` would still be `null` at this point, and if so, should we bail?